dali/kernels/slice/slice_gpu.cuh

```diff
@@ -261,11 +261,15 @@ class SliceGPU {
       total_volume += volume(args.shape);
     }
 
-    unsigned min_blocks = 4 * GetSmCount();
-    block_size_ = kMaxBlockSize;
-    while (total_volume / block_size_ < min_blocks && block_size_ > kMinBlockSize) {
-      block_size_ /= 2;
-    }
+    int blocksPerSM;
+    CUDA_CALL(cudaOccupancyMaxActiveBlocksPerMultiprocessor(&blocksPerSM,
+              detail::SliceKernel<OutputType, InputType, Dims, false>, kBlockDim, 0));
```

Comment:
What happens if you run a pipeline that requires a Slice with padding? Does it mean we schedule more work than we should?

Reply:
In other words: this PR won't improve performance of Slice with padding, but it won't make it worse

Reply:
Understood

Reply:
How about storing `blocksPerSM` as a member and setting it once per object lieftime? The value returned by this call is always the same (it can vary across devices, but it's quite safe to assume that a particular instance of a kernel object is bound to one device).

Reply:
You're right, changed in https://github.com/NVIDIA/DALI/pull/3604#issuecomment-1010407950